jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -200,13 +207,22 @@ public String formatClusterFilterString() throws InvalidURLException {
     public Cluster findClusterInPool(String filter) throws SQLException {
         try {
             Cluster suitableCluster = null;
+            Map<Cluster, Long> clusterLoads = new HashMap<>();
             for (Cluster response :
                     clusterControllerClient
                             .listClusters(params.projectId(), params.region(), filter)
                             .iterateAll()) {
-                // TODO: decide on how to pick a cluster among cluster pool
+                if (response.hasMetrics()
+                        && response.getMetrics().getYarnMetricsOrDefault(YARN_MEMORY, 0) != 0) {
+                    clusterLoads.put(response, response.getMetrics().getYarnMetricsOrDefault(YARN_MEMORY, 0));
```

Comment:
If you if statement is taking care of ensuring that the key is present, I think you can just use `getYarnMetrics(YARN_MEMORY)` instead of getOrDefault